### PR TITLE
Make cuda.get_array_module support Variable

### PR DESCRIPTION
**Pull request by muupan**
This PR will make `cuda.get_array_module` support `Variable` arguments so that it will return the array module of `Variable.data`. It will also add tests for `cuda.get_array_module`.

### CODE REVIEW

File: chainer/cuda.py [hunk 1]
```diff
@@ -555,6 +559,8 @@ def get_array_module(*args):
 
     """
     if available:
+        args = (arg.data if isinstance(arg, chainer.Variable) else arg
```
Comment (okuta) [inline chainer/cuda.py]: Please use list for performance.
`args = [arg.data if instance(arg, chainer.Variable) else arg for arg in args]`.

File: chainer/cuda.py [hunk 2]
```diff
@@ -543,8 +544,11 @@ def reduce(in_params, out_params, map_expr, reduce_expr, post_map_expr,
 def get_array_module(*args):
     """Gets an appropriate one from :mod:`numpy` or :mod:`cupy`.
 
-    This is almost equivalent to :func:`cupy.get_array_module`. The only
-    difference is that this function can be used even if CUDA is not available.
+    This is almost equivalent to :func:`cupy.get_array_module`. The differences
+    are that this function can be used even if CUDA is not available and that
+    it will return their data arrays' array module for
```
Comment (okuta) [inline chainer/cuda.py]: typo?

Comment (muupan) [inline chainer/cuda.py]: You mean `.` at line 551?

Comment (okuta) [inline chainer/cuda.py]: Sorry, I misunderstood.

**Comment by muupan** [comment]
Removed the unnecessary period and used list instead of tuple.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**